src/main/java/it/gov/pagopa/payhub/activities/connector/sendnotification/config/SendApisHolder.java

```diff
@@ -0,0 +1,59 @@
+package it.gov.pagopa.payhub.activities.connector.sendnotification.config;
+
+import it.gov.pagopa.payhub.activities.config.RestTemplateConfig;
+import it.gov.pagopa.pu.sendnotification.controller.ApiClient;
+import it.gov.pagopa.pu.sendnotification.controller.BaseApi;
+import it.gov.pagopa.pu.sendnotification.controller.generated.SendApi;
+import jakarta.annotation.PreDestroy;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class SendApisHolder {
+
+    private final RestTemplate restTemplate;
+    private final SendApiClientConfig clientConfig;
+    private final SendApi sendApi;
+
+    private final ThreadLocal<String> bearerTokenHolder = new ThreadLocal<>();
+
+    public SendApisHolder(
+        SendApiClientConfig clientConfig,
+        RestTemplateBuilder restTemplateBuilder
+    ) {
+        this.restTemplate = restTemplateBuilder.build();
+        this.clientConfig = clientConfig;
+        ApiClient apiClient = buildApiClient();
+
+        if (clientConfig.isPrintBodyWhenError()) {
+          restTemplate.setErrorHandler(RestTemplateConfig.bodyPrinterWhenError("SEND"));
```

Comment:
let's use the repo name, it could be interpreted as PagoPA SEND, instead we are calling our inner send-notification
```suggestion
          restTemplate.setErrorHandler(RestTemplateConfig.bodyPrinterWhenError("SEND_NOTIFICATION"));
```